Framework/PythonInterface/plugins/algorithms/SaveP2D.py

```diff
@@ -0,0 +1,195 @@
+            of.write('#Inst: ' + Data.getInstrument().getName() + ".prm\n")
+            binning = form.format(
+                Data.getDimension(0).getBinWidth()) + ' ' + form.format(
+                    Data.getDimension(1).getBinWidth()) + "\n"
+            of.write('#Binning: ddperp' + binning)
+            of.write('#Bank: 1\n')
+            of.write('#2theta   lambda   d-value   dp-value   counts\n')
+            # Get number of bins
+            ndp = Data.getDimension(1).getNBins()
+            # create progress reporter
+            pr = Progress(None, start=0.0, end=1.0, nreports=ndp)
+            last_dp = -1
+            last_d = -1
+
+            # iterate over all dPerpendicular cuts
+            for cdp in range(ndp):
+                dp = Data.getDimension(1).getX(cdp)
+                # skip if no new dPerpendicular cut
+                if (dp == last_dp):
+                    continue
+                # create value of bin center from left and right border
+                dp_center = (dp + Data.getDimension(1).getX(cdp + 1)) / 2.
+                last_dp = dp
+                pr.report("Processing")
+                print("{:4.0f}%".format(cdp * 100. / ndp))
+                # iterate over all dSpacing values for the selected dPerpendicular value
+                for cd in range(Data.getDimension(0).getNBins()):
+                    d = Data.dataX(cdp)[cd]
+                    # skip if d is the same as before
+                    if (d == last_d):
+                        break
+                    last_d = d
+                    Y = Data.dataY(cdp)[cd]
+                    # skip NaN values for intensity if option is activated
+                    if self.getPropertyValue('RemoveNaN') == '1':
+                        if math.isnan(Y):
+                            continue
+                    # skip intensities <= 0 if option is selected
+                    if self.getPropertyValue('RemoveNegatives') == '1':
+                        if Y <= 0:
+                            continue
+                    # calculate 2theta and lambda from d and dPerpendicular
+                    dsq = d**2
+                    lhkl = np.sqrt(4. * dsq - wo_real(4. * dsq - np.log(0.25 / dsq) - dp_center**2))
+                    thkl = 2. * np.arcsin(lhkl / 2. / d) / np.pi * 180.
+                    # skip Values outside of specified data ranges if the option is activated
+                    if self.getPropertyValue('CutData') == '1':
+                        if self.check_data_ranges(d, dp_center, lhkl, thkl):
+                            # write data into outfile after checking all data ranges positive
+                            of.write(lform.format(thkl, lhkl, d, dp_center, Y))
+                    else:
+                        # Write data into outfile
+                        of.write(lform.format(thkl, lhkl, d, dp_center, Y))
+            print('\n\nExported: ' + OutFile + '\n')
+
+
+mantid.api.AlgorithmFactory.subscribe(SaveP2D)
```

Comment:
Upon changing the headers, we can make this call
```suggestion
AlgorithmFactory.subscribe(SaveP2D)
```